tests/suites/test_suite_arc4.function

```diff
@@ -7,6 +7,28 @@
  * END_DEPENDENCIES
  */
 
+/* BEGIN_CASE depends_on:MBEDTLS_CHECK_PARAMS:!MBEDTLS_PARAM_FAILED_ALT */
+void mbedtls_arc4_invalid_param( )
+{
+    mbedtls_arc4_context ctx;
+    unsigned char key[128] = { 0 };
+
+    TEST_VALID_PARAM( mbedtls_arc4_free( NULL ) );
+
+    /* Check that parameter validation detect NULL parameters. */
+    TEST_INVALID_PARAM( mbedtls_arc4_init( NULL ) );
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( NULL, key, sizeof( key ) ) );
+    TEST_INVALID_PARAM( mbedtls_arc4_setup( &ctx, NULL, 42 ) );
```

Comment:
This is currently correct, but I would consider providing a valid, initialized context here, as the semantics of this test is: Provide: a valid context, NULL buffer and a valid key length. This is not true, and is correct only because we check for the invalid key, before we run into problems with uninitialized context.

However, I am not sure if this nuance is worth investing time, as it is a rather pedantic remark.

Reply:
I'd prefer to keep this in line with the other parameter validation PRs for now, so I'll not change anything unless you insist.